sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -873,5 +1063,277 @@ public virtual Task FlushInternalAsync(CancellationToken cancellationToken = def
         {
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        ///   Queries for the identifiers of the Event Hub partitions.
+        /// </summary>
+        ///
+        /// <param name="producer">The producer client instance to use for querying.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the query.</param>
+        ///
+        /// <returns>The set of identifiers for the Event Hub partitions.</returns>
+        ///
+        protected virtual async Task<string[]> ListPartitionIdsAsync(EventHubProducerClient producer,
+                                                                     CancellationToken cancellationToken) =>
+            await producer.GetPartitionIdsAsync(cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        ///   Performs the actions needed to initialize the <see cref="EventHubBufferedProducerClient" /> to begin accepting events to be
+        ///   enqueued and published.  Should this method be called while processing is active, no action is taken.
```

Comment:
nit: I think it is more clear
```suggestion
        ///   enqueued and published.  If this method is called while processing is active, no action is taken.
```